### PR TITLE
Add node to key

### PR DESCRIPTION
**Pull request by joostjager**
Fixes bug where two nodes connected to multiplexer share a channel and cause circuit key collisions.

Circuit format in logs: `circuit=02a508a76083ef51bd60c994f5756e16aaafa946a29886397f411f3da0b32aa6ea:4039605720580096:19`

### CODE REVIEW

File: set_id.go [hunk 1]
```diff
@@ -16,16 +17,20 @@ func (s SetID) String() string {
...
-func newSetID(keys []types.CircuitKey) SetID {
-	sortedKeys := make([]types.CircuitKey, len(keys))
+func newSetID(keys []types.HtlcKey) SetID {
+	sortedKeys := make([]types.HtlcKey, len(keys))
 	copy(sortedKeys, keys)
 
 	sort.Slice(sortedKeys, func(a, b int) bool {
-		if sortedKeys[a].ChanID == sortedKeys[b].ChanID {
-			return sortedKeys[a].HtlcID < sortedKeys[b].HtlcID
+		if sortedKeys[a].Node != sortedKeys[b].Node {
```
Comment (joostjager) [inline set_id.go:25]: Fixing the hash collision!

Comment (Torakushi) [reply set_id.go:25]: What happens if `len(keys)==1`?

we could still have the same set_id if we make a payment A-->B or B-->A, do we want this behaviour?

Comment (Torakushi) [reply set_id.go:25]: I mean, in case of self-payment:

NewSetID(nodeA,htlcID,channelID) == NewSetID(nodeB,htlcID,channelID)

Comment (joostjager) [reply set_id.go:25]: Oops, forgot to add the node key 😬

Comment (joostjager) [reply set_id.go:25]: Also changed it to use a writer, to avoid the indexing logic.

File: set_id_test.go [hunk 2]
```diff
@@ -4,26 +4,59 @@ import (
...
+	hash = newSetID(keys)
+	require.NotEqual(t, expectedHash, hash[:])
+
+	keys = testKeys()
 	keys[0].HtlcID = 5
 	hash = newSetID(keys)
 	require.NotEqual(t, expectedHash, hash[:])
 }
+
+// TestSetIDSelfPayments tests that two payments in different directions between
```
Comment (Torakushi) [inline set_id_test.go:46]: Awesome!

**Review by Torakushi — APPROVED** [review]
LGTM!

**Pull request merged**